controllers/chat/sendAnonymousMessage.js

```diff
@@ -80,6 +83,11 @@ const sendAnonymousMessage = async (req, res) => {
       if (isBlocked) {
         return res.status(403).json(responseError('This user does not want to receive messages'));
       }
+      channelMember = better_channel_member.filter(
+        (element) => element.user_id !== req.userId && !blockedIdsSet.has(element.user_id)
+      );
+
+      senderInfo = better_channel_member.find((element) => element.user_id === req.userId);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There's no null-check for `better_channel_member`. If it's `undefined` or `null`, this will cause a runtime error. Also, the `find` method returns `undefined` if no elements in the array satisfy the provided testing function. You should handle this case.
```diff
+ if (!better_channel_member) {
+   throw new Error('better_channel_member is undefined or null');
+ }
  channelMember = better_channel_member.filter(
    (element) => element.user_id !== req.userId && !blockedIdsSet.has(element.user_id)
  );
  
  senderInfo = better_channel_member.find((element) => element.user_id === req.userId);
+ if (!senderInfo) {
+   throw new Error('No matching sender info found');
+ }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->